### PR TITLE
[stable/3.0] upgrade: Cleanup the current progress at the end of upgrade

### PR DESCRIPTION
**Pull request by jsuchome**
(cherry picked from commit 92267107d69ba9ad26054c82e82003c4cc1adc07)

Backport of https://github.com/crowbar/crowbar-core/pull/1142

### CODE REVIEW

**Comment by Itxaka** [comment]
Argh, I went to fast, same comment as in https://github.com/crowbar/crowbar-core/pull/1143, this is not needed as a backport I think, as it would be using the crowbar cloud7 version by the time the upgrade is finished.

**Comment by jsuchome** [comment]
@Itxaka you are right it is not strictly needed, but we are trying to keep the library in the same state in cloud6 and cloud7

**Comment by Itxaka** [comment]
> you are right it is not strictly needed, but we are trying to keep the library in the same state in cloud6 and cloud7

Good enough for me. :+1:

File: crowbar_framework/lib/crowbar/upgrade_status.rb [hunk 1]
```diff
@@ -144,8 +144,12 @@ def end_step(success = true, errors = {})
         }
         progress[:steps][current_step][:errors] = errors unless errors.empty?
         if current_step == upgrade_steps_6_7.last && success
-          # Mark the end of the upgrade process
+          # Mark the end of the upgrade process and cleanup the progress
           FileUtils.rm_f running_file
+          progress[:current_substep] = :end_of_upgrade
+          progress[:current_substep_status] = :finished
+          progress[:current_nodes] = {}
+          progress[:current_node_action] = "finished"
```
Comment (rsalevsky) [inline crowbar_framework/lib/crowbar/upgrade_status.rb]: Just a question: Is it intentional to have symbols and strings mixed?

Comment (jsuchome) [inline crowbar_framework/lib/crowbar/upgrade_status.rb]: This is correct, yes. We normally have symbols, but `current_node_action` contains really strings (as you could see from its usage from the code ... master, not 3.0)

Comment (jsuchome) [inline crowbar_framework/lib/crowbar/upgrade_status.rb]: See e.g. https://github.com/crowbar/crowbar-core/blob/master/crowbar_framework/app/models/api/upgrade.rb#L1029

Comment (rsalevsky) [inline crowbar_framework/lib/crowbar/upgrade_status.rb]: Thanks for the information, makes sense now.

**Pull request merged**